arm-notificationhubs/2016-03-01/swagger/notificationhubs.json

```diff
@@ -1247,115 +1165,81 @@
   "definitions": {
     "CheckAvailabilityParameters": {
       "properties": {
-        "id": {
-          "readOnly": true,
-          "type": "string",
-          "description": "Resource Id"
-        },
-        "name": {
-          "type": "string",
-          "description": "Resource name"
-        },
-        "type": {
-          "readOnly": true,
-          "type": "string",
-          "description": "Resource type"
-        },
-        "location": {
-          "type": "string",
-          "description": "Resource location"
-        },
-        "tags": {
-          "type": "object",
-          "additionalProperties": {
-            "type": "string"
-          },
-          "description": "Resource tags"
-        },
-        "sku": {
-          "$ref": "#/definitions/Sku",
-          "description": "The sku of the created namespace"
-        },
         "isAvailiable": {
           "type": "boolean",
-          "description": "True if the name is available and can be used to create new Namespace/NotificationHub. Otherwise false."
+          "description": "True if the name is available and can be used to create new namespace/notification hub, otherwise false"
         }
       },
       "required": [
-        "name",
-        "location"
+        "name"
+      ],
+      "allOf": [
```

Comment:
@ysxu - Think about it logically. As a customer I have not created a namespace yet. I simply want to check whether the name for the namespace is available or not. Then why should the service expect from me to provide location, tags, and other info? If it hasn't been created yet. How can that be a resource?